### ISSUE TITLE
Nest cmake fails when Python 3 version of Cython is installed at /usr/local/bin

### ISSUE THREAD

**Issue by tclose** [Homebrew/homebrew-science#6477]
### Indicate troubleshooting steps taken

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Checked that a similar issue hasn't already been filed?
- [x] Looked up the software bug tracker if reporting a build error?
- [ ] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports / build errors:

```
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> Downloading https://github.com/nest/nest-simulator/archive/v2.14.0.tar.gz
Already downloaded: /Users/tclose/Library/Caches/Homebrew/nest-2.14.0.tar.gz
==> Downloading https://files.pythonhosted.org/packages/d4/0c/9840c08189e030873387a73b90ada981885010dd9aea134d6de30cd2
Already downloaded: /Users/tclose/Library/Caches/Homebrew/nest--homebrew-virtualenv-15.1.0.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/tmp/nest--homebrew-virtualenv-20171204-6519-1d73l6l
==> python -s /tmp/nest--homebrew-virtualenv-20171204-6519-1d73l6l/target/bin/virtualenv -p python /usr/local/Cellar/n
==> Downloading https://files.pythonhosted.org/packages/ee/2a/c4d2cdd19c84c32d978d18e9355d1ba9982a383de87d0fcb5928553d
Already downloaded: /Users/tclose/Library/Caches/Homebrew/nest--Cython-0.27.3.tar.gz
==> /usr/local/Cellar/nest/2.14.0/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/t
==> Downloading https://files.pythonhosted.org/packages/58/a5/0dc93c3ec33f4e281849523a5a913fa1eea9a3068acfa754d44d8810
Already downloaded: /Users/tclose/Library/Caches/Homebrew/nest--nose-1.3.7.tar.gz
==> /usr/local/Cellar/nest/2.14.0/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/t
==> cmake .. -DCMAKE_INSTALL_PREFIX:PATH=/usr/local/Cellar/nest/2.14.0 -Dwith-python=2 -DPYTHON_LIBRARY=/usr/local/Cel
Last 15 lines from /Users/tclose/Library/Logs/Homebrew/nest/05.cmake:
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2")
-- Found GSL: /usr/local/include (found version "2.4")
-- Found PythonInterp: /usr/local/bin/python2 (found suitable version "2.7.14", minimum required is "2")
-- Found PythonLibs: /usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/libpython2.7.dylib (found suitable exact version "2.7.14")
-- Found PythonInterp: /usr/local/bin/python2 (found version "2.7.14")
-- Found Cython: /usr/local/bin/cython
CMake Error at cmake/ProcessOptions.cmake:375 (message):
  Your Cython version is too old.  Please install newer version (0.19.2+)
Call Stack (most recent call first):
  CMakeLists.txt:149 (nest_process_with_python)



-- Configuring incomplete, errors occurred!
See also "/tmp/nest-20171204-6519-1jfphb9/nest-simulator-2.14.0/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/nest-20171204-6519-1jfphb9/nest-simulator-2.14.0/build/CMakeFiles/CMakeError.log".
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: cmake .. -DCMAKE_INSTALL_PREFIX:PATH=/usr/local/Cellar/nest/2.14.0 -Dwith-python=2 -DPYTHON_LIBRARY=/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/lib/libpython2.7.dylib -DPYTHON_INCLUDE_DIR=/usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/include/python2.7
```

#### Brief summary of issue:

When building Nest from source `--with-python` the Cmake step of the build fails if Cython has been installed for Python 3 at `/usr/local/bin` complaining of an insufficient version (even if the version of Cython is up to date). This occurs despite the formula installing a local version of cython and putting it first on the system path before Cmake is called (may be an upstream bug). 

#### The complete command that you issued:

`brew install --debug --build-from-source nest --with-python`

#### Gist with output of `brew doctor`:

`Your system is ready to brew.`

#### Link output by `brew gist-logs <formula>`:

https://gist.github.com/488608f82e8409f11f9404c9e612973b

#### Output of `brew config`:

```
HOMEBREW_VERSION: 1.3.8-107-g0ad42eb
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0ad42ebbcb735422612a7c6d4edc5ad88044dd4b
Last commit: 83 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 053f0572e6f5f72ba486fe0d714b10108b556d5c
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
HOMEBREW_ENV_FILTERING: 1
HOMEBREW_GIT_CONFIG_DEVELOPERMODE: true
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.3.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 9.0 build 900
Git: 2.13.6 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Perl: /usr/bin/perl
Python: /usr/local/opt/python/libexec/bin/python => /usr/local/Cellar/python/2.7.14/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Java: 1.8.0_152
macOS: 10.13.1-x86_64
Xcode: N/A
CLT: 9.1.0.0.1.1508540944
X11: 2.7.11 => /opt/X11
```

**Comment by tammoippen** [comment]
Should be fixed by #6482. Suggest closing.

**Comment by tclose** [comment]
Fixed
